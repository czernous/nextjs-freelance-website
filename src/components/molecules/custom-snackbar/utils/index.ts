import { ICustomSnackbarProps } from '@src/interfaces/components-props';
import { SetStateAction } from 'react';

export const updateSnackbarProps = async (
  r: Response,
  callback: (value: SetStateAction<ICustomSnackbarProps | null>) => void,
) => {
  if (!r) return;
  const severity =
    r.status === 200 || r.status === 204 || r.status === 201
      ? 'success'
      : 'error';
  const hasJson = r.headers.get('Content-Type')?.includes('application/json');
  /* istanbul ignore next */
  const statusMessageText = async (r: Response) => {
    switch (r.status) {
      case 204:
        return 'Successfully saved data';
      case 201:
        return 'Successfully created';
      default:
        return await r.text();
    }
  };

  let statusMsg;

  try {
    statusMsg =
      hasJson && r.headers.get('Method') === 'PUT'
        ? await r.json()
        : {
            message: await statusMessageText(r),
          };
  } catch (error) {
    /* istanbul ignore next */
    statusMsg = hasJson
      ? {
          message:
            statusMsg.message ||
            (r.status === 200
              ? 'Successfully saved data'
              : 'Error saving data'),
        }
      : { message: 'Error reading response as text' };
  }

  callback({
    severity,
    text: statusMsg?.message ?? statusMsg,
  });
};
