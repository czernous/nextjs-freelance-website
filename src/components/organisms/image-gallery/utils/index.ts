import { IImage } from '@src/interfaces';
import { Dispatch, SetStateAction } from 'react';

/* istanbul ignore next */
export const getImages = async (): Promise<IImage[]> => {
  const imageResponse = await fetch(
    `${new URL(
      '/api/blog-data',
      window.location.origin,
    )}?url=/images&method=GET`,
  );

  const r = await imageResponse?.json();

  return r?.data;
};

/* istanbul ignore next */
export const handleGalleryOpen = async (
  setImages: Dispatch<SetStateAction<IImage[] | null>>,
) => {
  const imageResponse = await fetch(
    `${new URL(
      '/api/blog-data',
      window.location.origin,
    )}?url=/images&method=GET`,
  );

  const imagesData = await imageResponse?.json();

  setImages(imagesData?.data);
};

/* istanbul ignore next */
export const handleImageDelete = async (imageId: string) => {
  const imageDeleteResponse = await fetch(
    `${new URL(
      '/api/blog-data',
      window.location.origin,
    )}?url=/images/${imageId}&method=DELETE`,
  );

  if (imageDeleteResponse?.status === 204) {
    // TODO: add toast notification upon delete
  }

  if (imageDeleteResponse?.status === 404) {
    alert(
      'This image does not exist, you have probably deleted it already. Please reopen the gallery',
    );

    // TODO: add toast notification upon delete
  }
};
