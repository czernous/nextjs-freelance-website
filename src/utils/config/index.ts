import getConfig from 'next/config';

export function getBackendUrl() {
  const {
    serverRuntimeConfig: { BACKEND_URL },
  } = getConfig();
  return BACKEND_URL;
}

export function getBackendApiKey() {
  const {
    serverRuntimeConfig: { API_KEY },
  } = getConfig();
  return API_KEY;
}
