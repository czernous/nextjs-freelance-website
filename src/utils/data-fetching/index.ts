import path from 'path';
import { promises as fs } from 'fs';
import getConfig from 'next/config';

const { serverRuntimeConfig } = getConfig();

export const getFilePath = (dir: string, filename: string, extension: string) =>
  `${path.join(
    serverRuntimeConfig.PROJECT_ROOT,
    dir,
  )}/${filename}.${extension}`;

export const getPageData = async <T>(filename: string) => {
  const fileContents = await fs.readFile(filename, 'utf8');
  const pageData: T = JSON.parse(fileContents);

  return pageData;
};

export const writePageData = async <T>(filename: string, filebody: T) => {
  const date = new Date();

  await fs.writeFile(
    filename,
    JSON.stringify(Object.assign({ updatedAt: date }, filebody)),
  );
};
