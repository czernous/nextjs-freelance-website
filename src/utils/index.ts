export * from './data-fetching';
export * from './testing';
