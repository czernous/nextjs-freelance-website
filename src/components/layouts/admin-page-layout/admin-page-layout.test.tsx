import { act, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';

import { axe, toHaveNoViolations } from 'jest-axe';

import React from 'react';

import AdminPageLayout from './index';

import { render } from '../../../utils/testing';

expect.extend(toHaveNoViolations);

describe('Admin Page Layout', () => {
  let layout;

  beforeEach(async () => {
    layout = render(<AdminPageLayout title="test" />);
  });

  it('passes axe audit', async () => {
    const { container } = layout;
    await act(async () => {
      expect(await axe(container)).toHaveNoViolations();
    });
  });

  it('is present in layout', async () => {
    const layout = await screen.findByTestId('admin-layout');
    await waitFor(() => expect(layout).toBeInTheDocument());
  });

  it('is rendered unchanged', () => {
    const { container } = layout;
    expect(container).toMatchSnapshot();
  });
});
