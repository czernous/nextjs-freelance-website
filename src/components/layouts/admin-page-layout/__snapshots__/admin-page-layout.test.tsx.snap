// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Page Layout is rendered unchanged 1`] = `
<div>
  <main
    class="dashboard"
    data-testid="admin-layout"
  >
    <div
      class="sidenav"
    >
      <a
        href="/admin/posts"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium icon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          title="posts"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6iv9kh-MuiSvgIcon-root"
            data-testid="ListAltOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 7h6v2h-6zm0 4h6v2h-6zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7zM20.1 3H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9zM19 19H5V5h14v14z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/admin/pages"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium icon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          title="pages"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6iv9kh-MuiSvgIcon-root"
            data-testid="AutoStoriesOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22.47 5.2c-.47-.24-.96-.44-1.47-.61v12.03c-1.14-.41-2.31-.62-3.5-.62-1.9 0-3.78.54-5.5 1.58V5.48C10.38 4.55 8.51 4 6.5 4c-1.79 0-3.48.44-4.97 1.2-.33.16-.53.51-.53.88v12.08c0 .58.47.99 1 .99.16 0 .32-.04.48-.12C3.69 18.4 5.05 18 6.5 18c2.07 0 3.98.82 5.5 2 1.52-1.18 3.43-2 5.5-2 1.45 0 2.81.4 4.02 1.04.16.08.32.12.48.12.52 0 1-.41 1-.99V6.08c0-.37-.2-.72-.53-.88zM10 16.62C8.86 16.21 7.69 16 6.5 16s-2.36.21-3.5.62V6.71C4.11 6.24 5.28 6 6.5 6c1.2 0 2.39.25 3.5.72v9.9zM19 .5l-5 5V15l5-4.5V.5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium icon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        title="images"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6iv9kh-MuiSvgIcon-root"
          data-testid="CollectionsOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4v12H8V4h12m0-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-8.5 9.67 1.69 2.26 2.48-3.1L19 15H9zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        href="/logout"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium icon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          title="logout"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6iv9kh-MuiSvgIcon-root"
            data-testid="LogoutOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m17 8-1.41 1.41L17.17 11H9v2h8.17l-1.58 1.58L17 16l4-4-4-4zM5 5h7V3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h7v-2H5V5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
    <section>
      <h1
        class="heading"
      >
        test
      </h1>
    </section>
    <div
      data-testid="image-gallery"
    />
  </main>
</div>
`;
